x/auth/ante/ante_test.go

```diff
@@ -354,12 +344,12 @@ func TestAnteHandlerMemoGas(t *testing.T) {
 	checkInvalidTx(t, anteHandler, ctx, tx, false, sdk.CodeOutOfGas)
 
 	// memo too large
-	fee = types.NewStdFee(9000, sdk.NewCoins(sdk.NewInt64Coin("atom", 0)))
+	fee = types.NewStdFee(50000, sdk.NewCoins(sdk.NewInt64Coin("atom", 0)))
```

Comment:
We consume that much more gas now?

Reply:
Most of the time no. This was an upperbound enforced by 2 tests `MemoWithGas` and `MultiSig`. It's not intuitively clear to me why these tests suddenly consume much more gas. Have them passing for now and will investigate further to try and move the upperbound back down

Reply:
A major increase in gas in unavoidable because each decorator may read/write the same data (whereas monolithic antehandler can read/write once per value). This doesn’t actually correlate with an increase in computation/disk IO because of caching, but the gasKV store doesn’t distinguish between whether something was cached or not for consuming gas

I do think there are still some optimizations that can be made on a per-decorator basis, but i think that work should be done in a series of future PRs